### PR TITLE
Refactor Collection background tasks to Remove session dependency

### PR DESCRIPTION
**Pull request by avirajsingh7**
## Summary

This PR refactors the create_collection and delete_collection background task execution to simplify dependencies and improve reliability in both production and tests.
Now each background task opens its own database Session when needed, ensuring clean transaction boundaries and avoiding reusing closed sessions.

## Checklist

Before submitting a pull request, please ensure that you mark these task.

- [x] Ran `fastapi run --reload app/main.py` or `docker compose up` in the repository root and test.
- [x] If you've fixed a bug or added code that is tested and has test cases.

## Notes

Please add here if any other information is required for the reviewer.

## Summary by CodeRabbit

- New Features
  - Clearer status updates and error messages during collection creation and deletion.

- Refactor
  - Background processing for collection create/delete now manages work more reliably and consistently.

- Bug Fixes
  - Fewer edge-case failures by ensuring collection state and error info are recorded reliably.

- Tests
  - Tests updated to better simulate background tasks and session behavior for more robust coverage.

Note: No changes to user-facing API behavior.

### CODE REVIEW

File: backend/app/api/routes/collections.py [hunk 1]
```diff
@@ -339,15 +345,17 @@ def do_delete_collection(
...
-        data = collection_crud.delete(collection, assistant)
-        logger.info(
-            f"[do_delete_collection] Collection deleted successfully | {{'collection_id': '{collection.id}'}}"
-        )
-        callback.success(data.model_dump(mode="json"))
+        with Session(engine) as session:
+            collection_crud = CollectionCrud(session, current_user.id)
+            collection = collection_crud.read_one(request.collection_id)
+            assistant = OpenAIAssistantCrud(client)
+            data = collection_crud.delete(collection, assistant)
```
Comment (avirajsingh7) [inline backend/app/api/routes/collections.py:364]: As call to openAI is tightly coupled with collection delete, session may be open untill collection deleted from openAI.
@kartpop @AkhileshNegi should we de-couple this from session? 
Not doing this right now because this may require changes in document endpoint also. or better question can be do we require background job to delete a collection?

**Review by kartpop — APPROVED** [review]

File: backend/app/api/routes/collections.py (hunk 1 above)
Comment (kartpop) [reply backend/app/api/routes/collections.py:364]: good point

if we are not doing this right now, at least put a TODO comment here in code in this PR itself so that we can revisit this later

**Review by kartpop — APPROVED** [review]
